src/posix/platform/radio_spinel.cpp

```diff
@@ -705,6 +719,20 @@ void RadioSpinel::UpdateFdSet(fd_set &aReadFdSet, fd_set &aWriteFdSet, int &aMax
         FD_SET(sockFd, &aWriteFdSet);
     }
 
+    if (mTxRadioTid != 0)
```

Comment:
Can we use `mState` being in `kStateTransmitting` state instead of `mTxRadioTid != 0` check here?

Reply:
Updated accordingly.